### PR TITLE
Add a tool to identify teams with no team managers defined in our mapping and team managers that are no longer there

### LINKED ISSUE #1347: Nag about team managers who are no longer active/disabled
#1342 is introducing a team_managers.json file to know who to nag in case a triage owner becomes inactive/disabled. We should also nag somebody in case a manager in the team_managers.json file leaves.

### LINKED ISSUE #1387: Nag about teams with no team managers defined in our mapping

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #1347 and fixes #1387

## Checklist

- [x] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results

### CODE REVIEW

**Review by suhaibmujahid — APPROVED** [review]
LGTM. I added just some cosmetic comments; nothing affects the functionality.

File: auto_nag/team_managers.py [hunk 1]
```diff
@@ -25,8 +26,13 @@ def _load_team_managers(self, filepath):
                 for team, manager in json.load(file).items()
             }
 
-    def get_team_manager(self, team_name):
+    def get_team_manager(
+        self, team_name: str, no_fallback: bool = False
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/team_managers.py]: Done in 826fd3b8996b40417696395bcaad497f053f89de.

File: auto_nag/scripts/vacant_team_manager.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        return vacant_teams
+
+    def get_email_data(self, date: str, bug_ids: List[int]) -> List[dict]:
+        return self.identify_vacant_teams()
+
+    def organize_nag(self, data: List[dict]) -> List[dict]:
+        return data
+
+    def get_query_url_for_components(self, components: List[str]) -> Optional[str]:
+        return None
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Done in 9cc2aa6ff9cf42d33a22d5bdfcbe6ec094c173bf.

File: auto_nag/scripts/vacant_team_manager.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            product_handler=product_handler,
+        ).wait()
+
+        return data
+
+    def nag_template(self) -> str:
+        return self.template()
+
+    def identify_vacant_teams(self) -> List[dict]:
+        # Filter out products and components that are not active
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Done in aaa60ae07c147b6f0c03d833c9afbf05850785b1.

File: auto_nag/scripts/vacant_team_manager.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+            vacant_teams.append(info)
+
+        return vacant_teams
+
+    def get_email_data(self, date: str, bug_ids: List[int]) -> List[dict]:
+        return self.identify_vacant_teams()
+
+    def organize_nag(self, data: List[dict]) -> List[dict]:
+        return data
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Removed in d64ef273913ba55b9fd9091d16996f649382200e.

File: auto_nag/scripts/vacant_team_manager.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+from typing import List, Optional, Set
+
+from libmozdata.bugzilla import BugzillaProduct
+
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.nag_me import Nag
+from auto_nag.team_managers import TeamManagers
+
+
+class TeamManagerVacant(BzCleaner, Nag):
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Removed in d64ef273913ba55b9fd9091d16996f649382200e.

File: auto_nag/scripts/vacant_team_manager.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        return self.identify_vacant_teams()
+
+    def organize_nag(self, data: List[dict]) -> List[dict]:
+        return data
+
+    def get_query_url_for_components(self, components: List[str]) -> Optional[str]:
+        return None
+
+    def get_cc(self) -> Set[str]:
+        return set()
```
[the remark this answers is not in the document]
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Removed in d64ef273913ba55b9fd9091d16996f649382200e.

File: auto_nag/scripts/vacant_team_manager.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.team_managers import TeamManagers
+
+
+class TeamManagerVacant(BzCleaner):
+    def __init__(self) -> None:
+        super(TeamManagerVacant, self).__init__()
+        self.query_url = None
+        self.inactive_weeks_number = self.get_config("inactive_weeks_number", 26)
```
Comment (suhaibmujahid) [inline auto_nag/scripts/vacant_team_manager.py:18]:
```suggestion
```
This is not used. We do not check activity.

**Review by suhaibmujahid — APPROVED** [review]

File: auto_nag/scripts/vacant_team_manager.py (hunk 7 above)
Comment (marco-c) [reply auto_nag/scripts/vacant_team_manager.py]: Removed in d838868542720999e659d2e9cb1fb2ee6c399e27.

**Review by suhaibmujahid — APPROVED** [review]

**Pull request merged**